src/api/webrpc/gateway.go

```diff
@@ -12,6 +12,11 @@ import (
 
 // Gatewayer provides interfaces for getting skycoin related info.
 type Gatewayer interface {
+	GetInputData(in cipher.SHA256) (*historydb.UxOut, error)
+	GetTransactionInputsData(tx *coin.Transaction) ([]*historydb.UxOut, error)
+	GetBlockInputsData(block *coin.Block) ([][]*historydb.UxOut, error)
+	GetSignedBlockInputsData(block *coin.SignedBlock) ([][]*historydb.UxOut, error)
+	GetSignedBlocksInputsData(blocks []coin.SignedBlock) ([][][]*historydb.UxOut, error)
```

Comment:
Keep interface as small as possible, have GetUxoutByID() in `Gatewayer`, move the `GetTransactionInputsData` `GetBlockInputsData` etc, to the place where you want to call.

Reply:
You would like this change to be made only in api/webrpc/gateway.go or also in gui/gateway.go.